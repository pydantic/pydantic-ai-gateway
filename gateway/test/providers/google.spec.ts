import { GoogleGenAI } from '@google/genai'
import { describe, expect } from 'vitest'
import { test } from '../setup'

describe('google', () => {
  test('should call google via gateway', async ({ gateway }) => {
    const { fetch, otelBatch } = gateway

    // The `authToken` is passed as `Authorization` header with the anthropic client.
    const client = new GoogleGenAI({
      apiKey: 'healthy',
      httpOptions: { baseUrl: 'https://example.com/google-vertex' },
    })

    const response = await client.models.generateContent({
      model: 'gemini-2.5-flash',
      contents: 'What is the capital of france?',
      config: { maxOutputTokens: 1024, topP: 0.95, topK: 1, temperature: 0.5, stopSequences: ['potato'] },
    })

    expect(response).toMatchSnapshot('llm')
    expect(otelBatch, 'otelBatch length not 1').toHaveLength(1)
    expect(JSON.parse(otelBatch[0]!).resourceSpans?.[0].scopeSpans?.[0].spans?.[0]?.attributes).toMatchSnapshot('span')
  })
})
