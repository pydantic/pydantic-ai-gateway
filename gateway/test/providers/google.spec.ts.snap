// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`google > should call anthropic via gateway > llm 1`] = `
{
  "content": [
    {
      "text": "The capital of France is Paris.",
      "type": "text",
    },
  ],
  "id": "msg_01Dk8FLtHYcVGoNi8DFz19uD",
  "model": "claude-sonnet-4-20250514",
  "role": "assistant",
  "stop_reason": "end_turn",
  "stop_sequence": null,
  "type": "message",
  "usage": {
    "cache_creation": {
      "ephemeral_1h_input_tokens": 0,
      "ephemeral_5m_input_tokens": 0,
    },
    "cache_creation_input_tokens": 0,
    "cache_read_input_tokens": 0,
    "input_tokens": 14,
    "output_tokens": 10,
    "pydantic_ai_gateway": {
      "cost_estimate": 0.00019199999999999998,
    },
    "service_tier": "standard",
  },
}
`;

exports[`google > should call anthropic via gateway > span 1`] = `
[
  {
    "key": "logfire.msg",
    "value": {
      "stringValue": "chat claude-sonnet-4-20250514",
    },
  },
  {
    "key": "logfire.json_schema",
    "value": {
      "stringValue": "{"type":"object","properties":{"gen_ai.operation.name":{"type":"string"},"gen_ai.request.model":{"type":"string"},"gen_ai.system":{"type":"string"},"gen_ai.request.max_tokens":{"type":"number"},"gen_ai.request.top_k":{},"gen_ai.request.top_p":{},"gen_ai.request.temperature":{},"gen_ai.request.stop_sequences":{},"gen_ai.request.seed":{},"gen_ai.response.finish_reasons":{},"gen_ai.response.id":{"type":"string"},"gen_ai.input.messages":{},"gen_ai.output.messages":{},"gen_ai.system_instructions":{},"http.response.status_code":{"type":"number"},"http.request.body.text":{"type":"string"},"http.response.body.text":{"type":"string"},"gen_ai.response.model":{"type":"string"},"gen_ai.usage.input_tokens":{"type":"number"},"gen_ai.usage.cache_read_tokens":{"type":"number"},"gen_ai.usage.cache_write_tokens":{"type":"number"},"gen_ai.usage.output_tokens":{"type":"number"},"gen_ai.usage.input_audio_tokens":{},"gen_ai.usage.cache_audio_read_tokens":{},"gen_ai.usage.output_audio_tokens":{}}}",
    },
  },
  {
    "key": "logfire.level_num",
    "value": {
      "intValue": 9,
    },
  },
  {
    "key": "gen_ai.operation.name",
    "value": {
      "stringValue": "chat",
    },
  },
  {
    "key": "gen_ai.request.model",
    "value": {
      "stringValue": "claude-sonnet-4-20250514",
    },
  },
  {
    "key": "gen_ai.system",
    "value": {
      "stringValue": "anthropic",
    },
  },
  {
    "key": "gen_ai.request.max_tokens",
    "value": {
      "intValue": 1024,
    },
  },
  {
    "key": "gen_ai.request.top_k",
    "value": {},
  },
  {
    "key": "gen_ai.request.top_p",
    "value": {},
  },
  {
    "key": "gen_ai.request.temperature",
    "value": {},
  },
  {
    "key": "gen_ai.request.stop_sequences",
    "value": {},
  },
  {
    "key": "gen_ai.request.seed",
    "value": {},
  },
  {
    "key": "gen_ai.response.finish_reasons",
    "value": {
      "arrayValue": {
        "values": [
          {
            "stringValue": "end_turn",
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.response.id",
    "value": {
      "stringValue": "msg_01Dk8FLtHYcVGoNi8DFz19uD",
    },
  },
  {
    "key": "gen_ai.input.messages",
    "value": {
      "arrayValue": {
        "values": [
          {
            "kvlistValue": {
              "values": [
                {
                  "key": "role",
                  "value": {
                    "stringValue": "user",
                  },
                },
                {
                  "key": "parts",
                  "value": {
                    "arrayValue": {
                      "values": [
                        {
                          "kvlistValue": {
                            "values": [
                              {
                                "key": "type",
                                "value": {
                                  "stringValue": "text",
                                },
                              },
                              {
                                "key": "content",
                                "value": {
                                  "stringValue": "What is the capital of France?",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.output.messages",
    "value": {
      "arrayValue": {
        "values": [
          {
            "kvlistValue": {
              "values": [
                {
                  "key": "role",
                  "value": {
                    "stringValue": "assistant",
                  },
                },
                {
                  "key": "parts",
                  "value": {
                    "arrayValue": {
                      "values": [
                        {
                          "kvlistValue": {
                            "values": [
                              {
                                "key": "type",
                                "value": {
                                  "stringValue": "text",
                                },
                              },
                              {
                                "key": "content",
                                "value": {
                                  "stringValue": "The capital of France is Paris.",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                },
                {
                  "key": "finish_reason",
                  "value": {
                    "stringValue": "end_turn",
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
  {
    "key": "gen_ai.system_instructions",
    "value": {},
  },
  {
    "key": "http.response.status_code",
    "value": {
      "intValue": 200,
    },
  },
  {
    "key": "http.request.body.text",
    "value": {
      "stringValue": "{"model":"claude-sonnet-4-20250514","max_tokens":1024,"messages":[{"role":"user","content":"What is the capital of France?"}]}",
    },
  },
  {
    "key": "http.response.body.text",
    "value": {
      "stringValue": "{"model":"claude-sonnet-4-20250514","id":"msg_01Dk8FLtHYcVGoNi8DFz19uD","type":"message","role":"assistant","content":[{"type":"text","text":"The capital of France is Paris."}],"stop_reason":"end_turn","stop_sequence":null,"usage":{"input_tokens":14,"cache_creation_input_tokens":0,"cache_read_input_tokens":0,"cache_creation":{"ephemeral_5m_input_tokens":0,"ephemeral_1h_input_tokens":0},"output_tokens":10,"service_tier":"standard","pydantic_ai_gateway":{"cost_estimate":0.00019199999999999998}}}",
    },
  },
  {
    "key": "gen_ai.response.model",
    "value": {
      "stringValue": "claude-sonnet-4-20250514",
    },
  },
  {
    "key": "gen_ai.usage.input_tokens",
    "value": {
      "intValue": 14,
    },
  },
  {
    "key": "gen_ai.usage.cache_read_tokens",
    "value": {
      "intValue": 0,
    },
  },
  {
    "key": "gen_ai.usage.cache_write_tokens",
    "value": {
      "intValue": 0,
    },
  },
  {
    "key": "gen_ai.usage.output_tokens",
    "value": {
      "intValue": 10,
    },
  },
  {
    "key": "gen_ai.usage.input_audio_tokens",
    "value": {},
  },
  {
    "key": "gen_ai.usage.cache_audio_read_tokens",
    "value": {},
  },
  {
    "key": "gen_ai.usage.output_audio_tokens",
    "value": {},
  },
]
`;
